bson-scala/build.gradle

```diff
@@ -18,7 +18,7 @@ description = "A Scala wrapper / extension to the bson library"
 archivesBaseName = 'mongo-scala-bson'
 
 dependencies {
-    compile project(':bson')
+    implementation project(path: ':bson', configuration: 'archives')
```

Comment:
`compile` is deprecated in favor of `implementation`. The `configuration` parameter allows the project to have access to the JAR to get the exported version number from the manifest.

Reply:
Did this change anything about the manifests, or is this change about getting rid of gradle deprecations warnings?  If the latter, please revert these changes and defer to a separate ticket/PR.  I don't have a problem doing this, but each PR should be focused on one issue if feasible.

Reply:
I have created the ticket JAVA-3574 to address the Gradle deprecation warnings. I have reverted all deprecation related changes.